server/src/main/java/org/elasticsearch/indices/IndexDescriptor.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.indices;
+
+import org.elasticsearch.cluster.metadata.Metadata;
+
+import java.util.List;
+
+/**
+ * An IndexDescriptor is an object that
```

Comment:
...that does what? I probably should have finished this sentence.

- [ ] finish writing this javadoc